pkg/vault-mon/pki.go

```diff
@@ -75,7 +84,8 @@ func (mon *PKIMon) Watch(interval time.Duration) {
 			for _, pki := range mon.pkis {
 				log.Infof("Refresh PKI certificate for %s", pki.path)
 				pki.clearCerts()
-				err := pki.loadCerts()
+
+				err := pki.loadCerts(loadCertsDuration)
```

Comment:
IMO you should observe the duration in this func rather than passing the histogram as an argument.

Alternatively, define the metric as scoped to the package instead of this func so you don't have to pass around the reference.

Reply:
0b3720f2d8430e896fa2729ee85885dac8fe7443